plugins/inputs/radius/radius.go

```diff
@@ -0,0 +1,104 @@
+package radius
+
+import (
+	"context"
+	_ "embed"
+	"net"
+	"sync"
+	"time"
+
+	"layeh.com/radius"
+	"layeh.com/radius/rfc2865"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Radius struct {
+	Servers         []string
+	Username        string
+	Password        string
+	Secret          string
+	ResponseTimeout config.Duration
+	Log             telegraf.Logger
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (n *Radius) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Radius) Description() string {
+	return "Test Radius authentication"
+}
+
+func (n *Radius) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	if len(n.Servers) == 0 {
+		n.Servers = []string{"127.0.0.1:1812"}
+	}
+	if n.ResponseTimeout < config.Duration(time.Second) {
+		n.ResponseTimeout = config.Duration(time.Second * 5)
+	}
```

Comment:
Please move this to a `func (n *Radius) Init() error` function. Might be also good to convert all `(n *Radius)` instances to `(r *Radius)`...